### PR TITLE
Proxy task 14: typescript migration + eslint

### PR DESCRIPTION
**Pull request by a-prakaptsova**

### CODE REVIEW

File: src/services/asteroidService.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import axios from 'axios';
+import { calculateRequestDates } from '../utils/dateHelper';
+import { Exception } from '../utils/exceptions/Exception';
+import { mapAsteroidsData } from '../utils/asteroidsMapper';
+import { config } from '../config/config';
+import type { AsteroidNasaData, AsteroidQueryParams, AsteroidResponse } from '../utils/types/asteroid';
+
+export const getAsteroidsInPeriod = async (params: AsteroidQueryParams): Promise<AsteroidResponse> => {
+    try {
+        const dates = calculateRequestDates(params.start_date, params.end_date);
```
Comment (MShachykau) [inline src/services/asteroidService.ts]: what the "dates" is? (naming)

File: src/services/roverPhotoService.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import axios, { AxiosError } from 'axios';
+import { Exception } from '../utils/exceptions/Exception';
+import { config } from '../config/config';
+import { type Manifest } from '../utils/types/rover';
+
+export const getMostRecentRoverPhotoLink = async (apiKey: string): Promise<string> => {
+    try {
+        const date = await getMostRecentRoverPhotoDate(apiKey);
+        const resp = await axios.get(config.nasaApi.getRoverPhotoUrl, {
```
Comment (MShachykau) [inline src/services/roverPhotoService.ts]: not informative variables

File: src/services/roverPhotoService.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            params: {
+                earth_date: date,
+                api_key: apiKey
+            }
+        });
+        return resp.data.photos.pop().img_src;
```
Comment (MShachykau) [inline src/services/roverPhotoService.ts]: what if the data will be undefined or the photos will be empty array?

File: src/services/roverPhotoService.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    } catch (err) {
+        return handleException(err);
+    }
+};
+
+const getMostRecentRoverPhotoDate = async (apiKey: string): Promise<string> => {
+    try {
+        const resp = await axios.get<Manifest>(config.nasaApi.getRoverManifestUrl, { params: { api_key: apiKey } });
+        return resp.data.photo_manifest.max_date;
```
Comment (MShachykau) [inline src/services/roverPhotoService.ts]: same

File: src/services/roverPhotoService.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        });
+        return resp.data.photos.pop().img_src;
+    } catch (err) {
+        return handleException(err);
+    }
+};
+
+const getMostRecentRoverPhotoDate = async (apiKey: string): Promise<string> => {
+    try {
+        const resp = await axios.get<Manifest>(config.nasaApi.getRoverManifestUrl, { params: { api_key: apiKey } });
```
Comment (MShachykau) [inline src/services/roverPhotoService.ts]: better use "res" or "response"

File: src/services/roverPhotoService.ts (hunk 3 above)
Comment (a-prakaptsova) [reply src/services/roverPhotoService.ts]: Fixed

File: src/services/asteroidService.ts (hunk 1 above)
Comment (a-prakaptsova) [reply src/services/asteroidService.ts]: All naming fixed

File: src/services/roverPhotoService.ts [hunk 6]
```diff
@@ -1,30 +1,37 @@
...
 const getMostRecentRoverPhotoDate = async (apiKey: string): Promise<string> => {
     try {
-        const resp = await axios.get<Manifest>(config.nasaApi.getRoverManifestUrl, { params: { api_key: apiKey } });
-        return resp.data.photo_manifest.max_date;
+        const roverManifest = await axios.get<Manifest>(config.nasaApi.getRoverManifestUrl, { params: { api_key: apiKey } });
+        const maxDate = roverManifest.data?.photo_manifest?.max_date;
+        if (!maxDate) {
+            throw new Exception(404, 'There is no data on the date of the last photo of the rover');
+        }
+        return roverManifest.data.photo_manifest.max_date;
```
Comment (MShachykau) [inline src/services/roverPhotoService.ts]: you can return just the maxDate.

**Review by MShachykau — APPROVED** [review]

**Pull request merged**